src/routes/v1/oauth/callback/post.js

```diff
@@ -30,34 +30,44 @@ export default async function(req, res, next) {
   // Get the provider module.
   const provider = await getClient(state.provider);
 
+  // Get claims fallback values if provided
+  const claimsMapping = provider.issuer.metadata.claimsMapping;
+  const fetchUserInfo = provider.issuer.metadata.fetchUserInfo;
+
   const tokenSet = await provider.authorizationCallback(null, req.body, {
     state: rawState,
     // Don't validate the nonce. Not great, but we don't store the nonce in a
     // session right now, so we can't validate this
     nonce: null
   });
 
-  const email = tokenSet.claims.email
-    ? tokenSet.claims.email.toLowerCase()
-    : tokenSet.claims.preferred_username
-    ? tokenSet.claims.preferred_username.toLowerCase()
-    : null;
+  const claims = tokenSet.claims;
+  let email = getClaim(claims, claimsMapping, "email");
+  const name = getClaim(claims, claimsMapping, "name");
+  const sub = getClaim(claims, claimsMapping, "sub");
```

Comment:
```suggestion
  const sub = claim.sub
```

Sub is a required field of OpenID connect, so this one shouldn't be needed.

Reply:
Only added this one, just in case someone had something funky here. If its guaranteed to not be different then we can revert it.